generators/typescript/sdk/client-class-generator/src/GeneratedWebsocketClientClassImpl.ts

```diff
@@ -0,0 +1,396 @@
+            name: "connect",
+            scope: Scope.Public,
+            isAsync: true,
+            parameters: [
+                {
+                    name: "args",
+                    type: `${this.serviceClassName}.${GeneratedWebsocketClientClassImpl.CONNECT_ARGS_INTERFACE_NAME}`,
+                    initializer: "{}"
+                }
+            ],
+            returnType: getTextOfTsNode(
+                ts.factory.createTypeReferenceNode(ts.factory.createIdentifier("Promise"), [
+                    this.getSocketTypeNode(context)
+                ])
+            ),
+            statements: statements.map(getTextOfTsNode)
+        });
+
+        context.sourceFile.addModule(serviceModule);
+        context.sourceFile.addClass(serviceClass);
+    }
+
+    private generateOptionsInterface(context: SdkContext): InterfaceDeclarationStructure {
+        const generatedEnvironments = context.environments.getGeneratedEnvironments();
+        return {
+            kind: StructureKind.Interface,
+            name: GeneratedWebsocketClientClassImpl.OPTIONS_INTERFACE_NAME,
+            properties: [
+                {
+                    name: GeneratedWebsocketClientClassImpl.ENVIRONMENT_OPTION_PROPERTY_NAME,
+                    type: getTextOfTsNode(
+                        context.coreUtilities.fetcher.Supplier._getReferenceToType(
+                            generatedEnvironments.getTypeForUserSuppliedEnvironment(context)
+                        )
+                    ),
+                    hasQuestionToken: generatedEnvironments.hasDefaultEnvironment(),
+                    docs: ["The environment to use for the websocket. Defaults to the default environment."]
+                },
+                {
+                    name: GeneratedWebsocketClientClassImpl.BASE_URL_OPTION_PROPERTY_NAME,
+                    type: getTextOfTsNode(
+                        context.coreUtilities.fetcher.Supplier._getReferenceToType(
+                            ts.factory.createKeywordTypeNode(ts.SyntaxKind.StringKeyword)
+                        )
+                    ),
+                    hasQuestionToken: this.channel.baseUrl != null,
+                    docs: ["Override the base URL of the websocket."]
+                },
+                {
+                    name: GeneratedWebsocketClientClassImpl.DEBUG_PROPERTY_NAME,
+                    type: getTextOfTsNode(ts.factory.createKeywordTypeNode(ts.SyntaxKind.BooleanKeyword)),
+                    hasQuestionToken: true,
+                    docs: ["Enable debug mode on the websocket. Defaults to false."]
+                },
+                {
+                    name: GeneratedWebsocketClientClassImpl.RECONNECT_ATTEMPTS_PROPERTY_NAME,
+                    type: getTextOfTsNode(ts.factory.createKeywordTypeNode(ts.SyntaxKind.NumberKeyword)),
+                    hasQuestionToken: true,
+                    docs: ["Number of reconnect attempts. Defaults to 30."]
```

Comment:
We'll attempt to reconnect _30_ times? That sounds like a lot, but Is this preserving the status quo?